### PR TITLE
[Refactoring] Moving some mailbox-api abstract test classes to interface

### PR DESCRIPTION
**Pull request by Arsnael**

### CODE REVIEW

File: mailbox/api/src/test/java/org/apache/james/mailbox/MailboxManagerStressContract.java [hunk 1]
```diff
@@ -41,45 +41,37 @@
...
 
-    protected abstract T provideManager() throws MailboxException;
-
-    protected abstract EventBus retrieveEventBus(T mailboxManager);
-
-    @BeforeEach
-    void setUp() throws Exception {
-        this.mailboxManager = provideManager();
-    }
+    EventBus retrieveEventBus(T mailboxManager);
```
Comment (chibenwa) [inline mailbox/api/src/test/java/org/apache/james/mailbox/MailboxManagerStressContract.java]: IMO we should let the underlying testing class give it's EventBus through a `EventBus getEventBus()` method.

File: mailbox/api/src/test/java/org/apache/james/mailbox/SubscriptionManagerContract.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+    Username USER1 = Username.of("test");
+    String MAILBOX1 = "test1";
+    String MAILBOX2 = "test2";
+    MailboxSession SESSION = MailboxSessionUtil.create(USER1);
+
+    SubscriptionManager getSubscriptionManager();
+
+    @AfterEach
+    default void teardown() throws SubscriptionException {
```
Comment (chibenwa) [inline mailbox/api/src/test/java/org/apache/james/mailbox/SubscriptionManagerContract.java]: Can we have a commit for getting rid of this aweful lifecycle management method in a contract? <3

Cassandra + memory have this lifecycle comming for free, all you need is clear the JPA tables...

File: mailbox/maildir/src/test/java/org/apache/james/mailbox/maildir/DomainUserMaildirMailboxManagerStressTest.java [hunk 3]
```diff
@@ -21,17 +21,29 @@
...
+
+    @BeforeEach
+    void setUp() {
+        this.mailboxManager = provideManager();
+    }
+
+    private StoreMailboxManager provideManager() {
         try {
             return MaildirMailboxManagerProvider.createMailboxManager("/%domain/%user", tmpFolder);
         } catch (Exception e) {
```
Comment (chibenwa) [inline mailbox/maildir/src/test/java/org/apache/james/mailbox/maildir/DomainUserMaildirMailboxManagerStressTest.java]: Throw this exception and inline this method in setUp.

File: mailbox/maildir/src/test/java/org/apache/james/mailbox/maildir/FullUserMaildirMailboxManagerStressTest.java [hunk 4]
```diff
@@ -21,17 +21,29 @@
...
+    public StoreMailboxManager getManager() {
+        return mailboxManager;
+    }
+
+    @BeforeEach
+    void setUp() {
+        this.mailboxManager = provideManager();
+    }
+
+    private StoreMailboxManager provideManager() {
```
Comment (chibenwa) [inline mailbox/maildir/src/test/java/org/apache/james/mailbox/maildir/FullUserMaildirMailboxManagerStressTest.java]: inline this, setUp can throw an exception

File: mailbox/maildir/src/test/java/org/apache/james/mailbox/maildir/MaildirSubscriptionManagerTest.java [hunk 5]
```diff
@@ -20,19 +20,31 @@
...
+
     @Override
-    protected SubscriptionManager createSubscriptionManager() {
+    public SubscriptionManager getSubscriptionManager() {
+        return subscriptionManager;
+    }
+
+    @BeforeEach
+    void setUp() {
+        subscriptionManager = createSubscriptionManager();
```
Comment (chibenwa) [inline mailbox/maildir/src/test/java/org/apache/james/mailbox/maildir/MaildirSubscriptionManagerTest.java]: inline this please

File: mailbox/maildir/src/test/java/org/apache/james/mailbox/maildir/UserMaildirMailboxManagerStressTest.java [hunk 6]
```diff
@@ -21,17 +21,29 @@
...
+    public StoreMailboxManager getManager() {
+        return mailboxManager;
+    }
+
+    @BeforeEach
+    void setUp() {
+        this.mailboxManager = provideManager();
+    }
+
+    private StoreMailboxManager provideManager() {
```
Comment (chibenwa) [inline mailbox/maildir/src/test/java/org/apache/james/mailbox/maildir/UserMaildirMailboxManagerStressTest.java]: inline this, throw the exception

File: mailbox/memory/src/test/java/org/apache/james/mailbox/inmemory/MemoryMailboxManagerStressTest.java [hunk 7]
```diff
@@ -19,19 +19,31 @@
...
-    
     @Override
-    protected InMemoryMailboxManager provideManager() {
+    public InMemoryMailboxManager getManager() {
+        return mailboxManager;
+    }
+
+    @BeforeEach
+    void setUp() {
+        this.mailboxManager = provideManager();
```
Comment (chibenwa) [inline mailbox/memory/src/test/java/org/apache/james/mailbox/inmemory/MemoryMailboxManagerStressTest.java]: inline this?

File: mailbox/memory/src/test/java/org/apache/james/mailbox/inmemory/manager/InMemoryQuotaMailboxMessageManagerTest.java [hunk 8]
```diff
@@ -19,19 +19,41 @@
...
-public class InMemoryQuotaMailboxMessageManagerTest extends QuotaMessageManagerTest<StoreMailboxManager> {
+class InMemoryQuotaMailboxMessageManagerTest implements QuotaMessageManagerContract<StoreMailboxManager> {
+
+    private ManagerTestProvisionner provisionner;
+
+    private  IntegrationResources<StoreMailboxManager> resources;
+
+    @BeforeEach
+    void setUp() throws Exception {
+        this.resources = createResources();
```
Comment (chibenwa) [inline mailbox/memory/src/test/java/org/apache/james/mailbox/inmemory/manager/InMemoryQuotaMailboxMessageManagerTest.java]: inline this?

File: mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/CassandraMailboxManagerStressTest.java [hunk 9]
```diff
@@ -21,28 +21,40 @@
...
+        this.mailboxManager = provideManager();
     }
 
     @Override
-    protected EventBus retrieveEventBus(CassandraMailboxManager mailboxManager) {
+    public EventBus retrieveEventBus(CassandraMailboxManager mailboxManager) {
         return mailboxManager.getEventBus();
     }
+
+    private CassandraMailboxManager provideManager() {
```
Comment (chibenwa) [inline mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/CassandraMailboxManagerStressTest.java]: inline this?

File: mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/CassandraSubscriptionManagerTest.java [hunk 10]
```diff
@@ -47,20 +47,32 @@
...
+
     @Override
-    protected SubscriptionManager createSubscriptionManager() {
+    public SubscriptionManager getSubscriptionManager() {
+        return subscriptionManager;
+    }
+
+    @BeforeEach
+    void setUp() {
+        subscriptionManager = createSubscriptionManager();
```
Comment (chibenwa) [inline mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/CassandraSubscriptionManagerTest.java]: inline this?

File: mailbox/jpa/src/test/java/org/apache/james/mailbox/jpa/JPASubscriptionManagerTest.java [hunk 11]
```diff
@@ -21,19 +21,36 @@
...
+
     @Override
-    protected SubscriptionManager createSubscriptionManager() {
+    public SubscriptionManager getSubscriptionManager() {
+        return subscriptionManager;
+    }
+
+    @BeforeEach
+    void setUp() {
+        subscriptionManager = createSubscriptionManager();
```
Comment (chibenwa) [inline mailbox/jpa/src/test/java/org/apache/james/mailbox/jpa/JPASubscriptionManagerTest.java]: indem?

File: mailbox/jpa/src/test/java/org/apache/james/mailbox/jpa/JpaMailboxManagerStressTest.java [hunk 12]
```diff
@@ -22,26 +22,36 @@
...
+
     @Override
-    protected OpenJPAMailboxManager provideManager() {
+    public OpenJPAMailboxManager getManager() {
+        return openJPAMailboxManager.get();
+    }
+
+    @BeforeEach
+    void setUp() {
+        this.openJPAMailboxManager = provideManager();
```
Comment (chibenwa) [inline mailbox/jpa/src/test/java/org/apache/james/mailbox/jpa/JpaMailboxManagerStressTest.java]: idem?

File: mailbox/api/src/test/java/org/apache/james/mailbox/SubscriptionManagerContract.java (hunk 2 above)
Comment (Arsnael) [reply mailbox/api/src/test/java/org/apache/james/mailbox/SubscriptionManagerContract.java]: I thought I removed that... but I had to do a rebase before pushing, maybe it got messed up. Will fix

**Comment by Arsnael** [comment]
```
[87537fb4460afd40e96e9397b3b6cf6f2bb93b42] [ERROR]   ConcurrentTestRunnerTest.runShouldPerformAllOperationsEvenOnOccasionalExceptions:230 » NotTerminated
```
test this please

**Comment by Arsnael** [comment]
```
[87537fb4460afd40e96e9397b3b6cf6f2bb93b42] [ERROR] Failed to execute goal on project james-server-util: Could not resolve dependencies for project org.apache.james:james-server-util:jar:3.5.0-SNAPSHOT: Failed to collect dependencies at com.github.steveash.guavate:guavate:jar:1.0.0 -> com.google.guava:guava:jar:25.1-jre -> org.codehaus.mojo:animal-sniffer-annotations:jar:1.14: Failed to read artifact descriptor for org.codehaus.mojo:animal-sniffer-annotations:jar:1.14: Could not transfer artifact org.codehaus.mojo:mojo-parent:pom:34 from/to central (https://repo.maven.apache.org/maven2): /root/.m2/repository/org/codehaus/mojo/mojo-parent/34/mojo-parent-34.pom.part (No such file or directory) -> [Help 1]
```
test this please

**Comment by Arsnael** [comment]
```
[ERROR]   ConcurrentTestRunnerTest.runShouldPerformAllOperationsEvenOnOccasionalExceptions:230 » NotTerminated
```
weird that it pops twice in 30 minutes... will check a bit

**Comment by chibenwa** [comment]
test this please

**Comment by Arsnael** [comment]
```
[87537fb4460afd40e96e9397b3b6cf6f2bb93b42] [ERROR]   ConcurrentTestRunnerTest.runShouldPerformAllOperationsEvenOnOccasionalExceptions:230 » NotTerminated
```
again... I do not have this issue on local though... Seems the last few builds were from `ci-james-01` when I checked

**Comment by Arsnael** [comment]
test this please

**Comment by Arsnael** [comment]
```
[87537fb4460afd40e96e9397b3b6cf6f2bb93b42] [ERROR] Failures: 
[87537fb4460afd40e96e9397b3b6cf6f2bb93b42] [ERROR]   MaildirSubscriptionManagerTest.subscribeShouldBeIdempotent 
[87537fb4460afd40e96e9397b3b6cf6f2bb93b42] Expecting:
[87537fb4460afd40e96e9397b3b6cf6f2bb93b42]   <["test1", "test2"]>
[87537fb4460afd40e96e9397b3b6cf6f2bb93b42] to contain exactly (and in same order):
[87537fb4460afd40e96e9397b3b6cf6f2bb93b42]   <["test1"]>
[87537fb4460afd40e96e9397b3b6cf6f2bb93b42] but some elements were not expected:
[87537fb4460afd40e96e9397b3b6cf6f2bb93b42]   <["test2"]>
[87537fb4460afd40e96e9397b3b6cf6f2bb93b42] 
[87537fb4460afd40e96e9397b3b6cf6f2bb93b42] [ERROR]   MaildirSubscriptionManagerTest.user1ShouldBeAbleToSubscribeOneMailbox 
[87537fb4460afd40e96e9397b3b6cf6f2bb93b42] Expecting:
[87537fb4460afd40e96e9397b3b6cf6f2bb93b42]   <["test1", "test2"]>
[87537fb4460afd40e96e9397b3b6cf6f2bb93b42] to contain exactly (and in same order):
[87537fb4460afd40e96e9397b3b6cf6f2bb93b42]   <["test1"]>
[87537fb4460afd40e96e9397b3b6cf6f2bb93b42] but some elements were not expected:
[87537fb4460afd40e96e9397b3b6cf6f2bb93b42]   <["test2"]>
```
Fixing this...

**Comment by Arsnael** [comment]
```
05:35:53.364 [ERROR] spark.Spark - ignite failed
```
test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**